Bus-Updater/src/update.cpp

```diff
@@ -276,19 +280,36 @@ static unsigned char updUnlockDevice(uint8_t * data)
         // as a simple method we use the unique ID of the CPU itself
         // only if this UID (GUID) is known, the device will be unlocked
         byte uid[IAP_UID_LENGTH];
-        lastError = (UDP_State)iapReadUID(uid);
+        lastError = iapResult2UDPState(iapReadUID(uid));
         if (lastError != UDP_IAP_SUCCESS)
         {
             // could not read UID of mcu
-            setLastError((UDP_State)lastError);
+            setLastError(lastError);
             return (T_ACK_PDU);
         }
 
-        for (unsigned int i = 0; i < UID_LENGTH_USED; i++)
+        if (memcmp(uid, data, min(size, UID_LENGTH_USED)) != 0)
         {
-            if (data[i + 3] != uid[i])
+            if (UID_LENGTH_USED % sizeof(uint32_t) != 0)
```

Comment:
valid values for `UID_LENGTH_USED` could theoretically range from 1 to 16. This check only guards the later byte "reversion", so we can accept uids in "Flashmagic-format". Right now we use 12 bytes. With this PR's protocol change we could increase it to 13, later with extened frames even to 16. A compile-time check restricts us to 4, 8, 12 or 16, but ensures "Flashmagic-Format" conversion. For me that restriction would be ok. Do you have any advice to implement this check at compile time?

Reply:
If you want to enforce it at compile time, it would be written like this:

```cpp
static_assert(UID_LENGTH_USED % sizeof(uint32_t) == 0, "UID_LENGTH_USED must be multiple of sizeof(uint32_t)");
```

Reply:
addressed in 8b56654. Thank you. From now on I'm gonna use `static_assert` thenever I can.